test/helper.rb

```diff
@@ -63,36 +70,61 @@ def get_adapter_name
 class FauxWriter < Datadog::Writer
   def initialize
     super(transport: FauxTransport.new(HOSTNAME, PORT))
+    @mutex = Mutex.new
 
     # easy access to registered components
     @spans = []
     @services = {}
   end
 
   def write(trace, services)
-    super(trace, services)
-    @spans << trace
-    @services = services
+    @mutex.synchronize do
+      super(trace, services)
+      @spans << trace
+      @services = services
+    end
   end
 
   def spans
-    spans = @spans
-    @spans = []
-    spans.flatten
+    @mutex.synchronize do
+      spans = @spans
+      @spans = []
+      spans.flatten!
+      # sort the spans to avoid test flakiness
+      spans.sort! do |a, b|
```

Comment:
Oh ok, removed previous comments because I wasn't aware of that. This is a test so everything that is fine to make testing easier and reliable 👍